src/main/scala/centaur/test/Test.scala

```diff
@@ -100,28 +98,37 @@ object Operations {
   }
 
   def validateMetadata(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata): Test[Unit] = {
-    val consistencyTimeout = Future { Thread.sleep(CentaurConfig.metadataConsistencyTimeout.toMillis) }
+
+    def eventually(startTime: OffsetDateTime, timeout: FiniteDuration)(f: => Try[Unit]): Try[Unit] = {
```

Comment:
The way this is written kind of looks tail recursive but it's not.  It's easy enough to revise this into a tail recursive `match` if that's important.